source/material_model/modified_tait.cc

```diff
@@ -0,0 +1,288 @@
+          const double psubpth = pressure - reference_pressure - Pth_rel;
+
+          // x = rho0/rho
+          const double x = 1 - a * (1. - std::pow((1. + b * psubpth), -1.0 * c));
+
+          // xi
+          double xi = (C_V / C_V0);
+
+          // Here we calculate the pressure effect on the heat capacity. It is a bit involved.
+          const double dintVdpdT = (reference_thermal_expansivity * reference_isothermal_bulk_modulus / reference_rho * a * xi) * (
+                                     std::pow((1. + b * psubpth), - c) - std::pow((1. - b * Pth), - c));
+
+          const double dSdT = reference_isothermal_bulk_modulus / reference_rho * std::pow((xi * reference_thermal_expansivity), 2) * \
+                              (std::pow((1. + b * psubpth), -1. - c) - std::pow((1. - b * Pth), -1. - c)) + \
+                              dintVdpdT * (( 1 - 2./x + 2./(std::exp(x) - 1.) ) * einstein_temperature/(temperature*temperature));
+
+
+          const Point<1> Tpoint(temperature);
+          double rho = reference_rho / x;
+          double isothermal_compressibility = 1./(reference_isothermal_bulk_modulus * (1. + b * psubpth) * (a + (1. - a) * std::pow((1. + b * psubpth), c)));
+          double alpha = reference_thermal_expansivity * xi * 1. / ((1. + b * psubpth) * (a + (1. - a) * std::pow((1 + b * psubpth), c)));
+          double heat_capacity = reference_heat_capacity_function.value(Tpoint) + temperature * dSdT;
+
+          // The following line might be useful for some compressibility models
+          // double isentropic_compressibility = isothermal_compressibility - alpha*alpha*temperature/(rho*heat_capacity);
+          out.densities[i] = rho;
+          out.compressibilities[i] = isothermal_compressibility;
+          out.thermal_expansion_coefficients[i] = alpha;
+          out.specific_heat[i] = heat_capacity;
+
+          out.entropy_derivative_pressure[i] = 0.0;
+          out.entropy_derivative_temperature[i] = 0.0;
+          // Change in composition due to chemical reactions at the
+          // given positions. The term reaction_terms[i][c] is the
+          // change in compositional field c at point i.
+          for (unsigned int c=0; c<in.composition[i].size(); ++c)
+            out.reaction_terms[i][c] = 0.0;
+        }
+    }
+
+    template <int dim>
+    double
+    ModifiedTait<dim>::
+    reference_viscosity () const
+    {
+      return eta;
+    }
+
+
+
+    template <int dim>
+    bool
+    ModifiedTait<dim>::
+    is_compressible () const
+    {
+      return true;
+    }
+
+
+
+    template <int dim>
+    void
+    ModifiedTait<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Material model");
+      {
+        prm.enter_subsection("Modified Tait model");
+        {
+          prm.declare_entry ("Reference pressure", "1e5",
+                             Patterns::Double (0),
+                             "Reference pressure $\\P_0$. Units: $Pa$.");
```

Comment:
The `\\` in front of the `P_0` looks wrong. In principle, the `Pa` should be typeset as `\\text{Pa}` (as I do in #3044, for example), but I won't hold you to actually doing that if you don't want.

Same below for the parameters that follow.